### PR TITLE
Addition of migration id and migration status in get server api

### PR DESCRIPTION
**Pull request by YashviJain01**
### :nut_and_bolt: Description: What code changed, and why?

Added Migration id and migration status for active migration in get server API to show the status on Server page itself.

### :chains: Related Resources

Modify the Get Server API to provide active migration id and status #6561

### :+1: Definition of Done

Migration id and migration status can be fetched in the server page now. If a server has a active migration going on, the migration id and migration type column will be fetched

### :athletic_shoe: How to Build and Test the Change

rebuild components/automate-gateway/
rebuild components/infra-proxy-service

Ensure Chef Infra Server is running. If it is not start it with start_chef_server
Add the chef-server test environment credentials and extract the SERVER IDs

Request: 
``` curl --location --request GET 'https://a2-dev.test/api/v0/infra/servers/${server_id}' \--header 'api-token: ${get_admin_token} ```

Response:
``` {
    "server": {
        "id": "test",
        "name": "test",
        "fqdn": "fqdn",
        "ip_address": "",
        "orgs_count": 0,
        "migration_id": "id",
        "migration_status": "status"
    }
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [X] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [X] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [X] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [X] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [X] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [X] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [X] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/infra-proxy-service/storage/postgres/migrations.go [hunk 1]
```diff
@@ -143,3 +142,18 @@ func (p *postgres) insertMigration(ctx context.Context, migrationId, serverId, m
 
 	return m, nil
 }
+
+//GetActiveMigration gets the Migration ID and Migration Status for a server id
+func (p *postgres) GetActiveMigration(ctx context.Context, serverId string) (storage.ActiveMigration, error) {
+	var m storage.ActiveMigration
+	query := "select m.id,t.type from migration m join migration_type t on m.type_id=t.id and t.id < 5000 and m.server_id=$1;"
```
Comment (kalroy) [inline components/infra-proxy-service/storage/postgres/migrations.go:149]: We need to make sure that we are picking up the latest status and the migration id. We can make sure that only one migration occurs for a server but we should be fetching the latest status ordered by the updated timestamp.

**Review by kalroy — CHANGES_REQUESTED** [review]

**Review by kalroy — CHANGES_REQUESTED** [review]
Add the unit test for these changes

**Comment by YashviJain01** [comment]
> Add the unit test for these changes

Created a issue for this change, we can pick this uo
https://github.com/chef/automate/issues/6620

**Review by vinay033 — APPROVED** [review]
LGTM

File: components/infra-proxy-service/storage/postgres/migrations.go [hunk 2]
```diff
@@ -145,6 +144,21 @@ func (p *postgres) insertMigration(ctx context.Context, migrationId, serverId, m
 	return m, nil
 }
 
+//GetActiveMigration gets the Migration ID and Migration Status for a server id
+func (p *postgres) GetActiveMigration(ctx context.Context, serverId string) (storage.ActiveMigration, error) {
+	var m storage.ActiveMigration
+	query := "select m.id,t.type from migration m join migration_type t on m.type_id=t.id and t.id < 5000 and m.server_id=$1 order by updated_timestamp desc FETCH FIRST ROW ONLY"
```
Comment (sonali523) [inline components/infra-proxy-service/storage/postgres/migrations.go:150]: Are we not dealing with the migration_status here as In Progress,completed or failed?

Comment (YashviJain01) [reply components/infra-proxy-service/storage/postgres/migrations.go:150]: no, as in the migration _type table the statuses are clearly mentioned

Comment (sonali523) [reply components/infra-proxy-service/storage/postgres/migrations.go:150]: okay

**Review by sonali523 — APPROVED** [review]

File: components/infra-proxy-service/storage/postgres/migrations.go (hunk 2 above)
Comment (kalroy) [reply components/infra-proxy-service/storage/postgres/migrations.go:150]: I am assuming that if the migration is not completed or failed then we should be in any of the phases and in progress

**Review by kalroy — APPROVED** [review]

**Pull request merged**